.github/workflows/release.yaml

```diff
@@ -0,0 +1,76 @@
+name: Release
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+
+      - name: Install tools
+        run: |
+          sudo apt-get install moreutils
+          pip install crudini
+
+      - name: Export current snapshot of app
+        run: echo "SDLAPP_VERSION_CURRENT_SNAPSHOT=$(jq -r '.meta.version' globalConfig.json)" >> $GITHUB_ENV
+
+      - name: Export release version of app
+        run: echo "SDLAPP_VERSION_RELEASE=`echo '${{ env.SDLAPP_VERSION_CURRENT_SNAPSHOT }}' | sed -En 's/-SNAPSHOT//p'`" >> $GITHUB_ENV
+
+      - name: Export new snapshot of app
+        run: echo "SDLAPP_VERSION_NEW_SNAPSHOT=`awk -vFS=. -vOFS=. '{$NF++;print}' <<<'${{ env.SDLAPP_VERSION_RELEASE }}'`-SNAPSHOT" >> $GITHUB_ENV
+
+      - name: Build release
+        run: |
+          python3.8 -m venv venv
+          source venv/bin/activate
+          pip install "cython<3.0.0"
+          pip install wheel
+          pip install --no-build-isolation pyyaml==5.4.1
+          pip install --upgrade-strategy only-if-needed splunk-appinspect
+          pip install --upgrade-strategy only-if-needed splunk-add-on-ucc-framework
+          pip install --upgrade-strategy only-if-needed splunk-packaging-toolkit
+          ucc-gen build --source TA_dataset --ta-version ${{ env.SDLAPP_VERSION_RELEASE }}
+          slim package output/TA_dataset -o release
+          
+      - name: Prepare next snapshot
+        run: |
+          jq ".meta.version=\"${{ env.SDLAPP_VERSION_NEW_SNAPSHOT }}\"" globalConfig.json | sponge globalConfig.json
+          jq ".info.id.version=\"${{ env.SDLAPP_VERSION_NEW_SNAPSHOT }}\"" TA_dataset/app.manifest | sponge TA_dataset/app.manifest
+          crudini --set TA_dataset/default/app.conf launcher version ${{ env.SDLAPP_VERSION_NEW_SNAPSHOT }}
+          crudini --set TA_dataset/default/app.conf id version ${{ env.SDLAPP_VERSION_NEW_SNAPSHOT }}
+          sed -i 's/${{ env.SDLAPP_VERSION_CURRENT_SNAPSHOT }}/${{ env.SDLAPP_VERSION_NEW_SNAPSHOT }}/' README.md
+
+      - name: Commit version changes
+        uses: test-room-7/action-update-file@v1
```

Comment:
Where we reference public non-audited actions from the marketplace we should pin them to a specific git commit sha which corresponds to v1 tag (e.g. https://github.com/scalyr/scalyr-agent-2/blob/master/.github/workflows/codespeed-benchmarks.yml#L27).

Having said that, this action also looks not so trustworthy - https://github.com/test-room-7/action-update-file. It only has a couple of starts and forks so we should see if there are any other similar more popular actions.

Reply:
Here is an example of how we do a similar thing in another repo directly using bash without any actions - https://github.com/scalyr/dataset-exporter-opentelemetry/blob/main/.github/workflows/benchmarks.yaml#L148